core/components/ui/gallery/gallery.tsx

```diff
@@ -17,282 +16,117 @@ interface Image {
   altText: string;
 }
 
-const GalleryContext = createContext<{
+interface Props extends ComponentPropsWithoutRef<'div'> {
   images: Image[] | [];
-  selectedImageIndex: number;
-  setSelectedImageIndex: React.Dispatch<React.SetStateAction<number>>;
-}>({
-  images: [],
-  selectedImageIndex: 0,
-  setSelectedImageIndex: () => null,
-});
-
-const ThumbnailContext = createContext<{
-  index: number;
-}>({
-  index: 0,
-});
-
-const GalleryPreviousIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWithRef<'button'>>(
-  ({ children, className, onClick, ...props }, ref) => {
-    const { images, selectedImageIndex, setSelectedImageIndex } = useContext(GalleryContext);
-    const previousIndex = selectedImageIndex === 0 ? images.length - 1 : selectedImageIndex - 1;
-
-    return (
-      <button
-        aria-label="Previous product image"
-        className={cn(
-          'focus-visible:outline-none focus-visible:ring-4 focus-visible:ring-primary/20',
-          className,
-        )}
-        onClick={(e) => {
-          setSelectedImageIndex(previousIndex);
-
-          if (onClick) {
-            onClick(e);
-          }
-        }}
-        ref={ref}
-        {...props}
-      >
-        {children || <ChevronLeft />}
-      </button>
-    );
-  },
-);
-
-GalleryPreviousIndicator.displayName = 'GalleryPreviousIndicator';
-
-const GalleryNextIndicator = forwardRef<ElementRef<'button'>, ComponentPropsWithRef<'button'>>(
-  ({ children, className, onClick, ...props }, ref) => {
-    const { images, selectedImageIndex, setSelectedImageIndex } = useContext(GalleryContext);
-    const nextIndex = selectedImageIndex + 1 < images.length ? selectedImageIndex + 1 : 0;
-
-    return (
-      <button
-        aria-label="Next product image"
-        className={cn(
-          'focus-visible:outline-none focus-visible:ring-4 focus-visible:ring-primary/20',
-          className,
-        )}
-        onClick={(e) => {
-          setSelectedImageIndex(nextIndex);
-
-          if (onClick) {
-            onClick(e);
-          }
-        }}
-        ref={ref}
-        {...props}
-      >
-        {children || <ChevronRight />}
-      </button>
-    );
-  },
-);
-
-GalleryNextIndicator.displayName = 'GalleryNextIndicator';
-
-const GalleryControls = forwardRef<ElementRef<'div'>, ComponentPropsWithRef<'div'>>(
-  ({ children, className, ...props }, ref) => {
-    const { images } = useContext(GalleryContext);
-
-    if (images.length <= 1) {
-      return null;
-    }
+  defaultImageIndex?: number;
+  noImageText?: string;
+}
 
-    return (
-      <div
-        className={cn(
-          'absolute top-1/2 flex w-full -translate-y-1/2 justify-between px-5 sm:px-0',
-          className,
+const Gallery = ({
+  className,
+  children,
+  images,
+  defaultImageIndex = 0,
+  noImageText,
```

Comment:
Remove prop, if someone really wants to change no-image-text they can just edit the component